File: CS/ButtonEditWithPadding/PaddingButtonEdit.cs
using System.ComponentModel;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Drawing;

namespace ButtonEditWithPadding
{
    public class PaddingButtonEdit : ButtonEdit
    {
        static PaddingButtonEdit()
        {
            RepositoryItemPaddingButtonEdit.Register();
        }
        public PaddingButtonEdit()
        {
        }

        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public new RepositoryItemPaddingButtonEdit Properties
        {
            get { return base.Properties as RepositoryItemPaddingButtonEdit; }
        }

        public override string EditorTypeName
        {
            get { return RepositoryItemPaddingButtonEdit.EditorName; }
        }


        protected override void OnClickButton(EditorButtonObjectInfoArgs buttonInfo)
        {
            ShowPopupForm();
            base.OnClickButton(buttonInfo);
        }
        protected virtual void ShowPopupForm()
        {
            using ( Form form = new Form() )
            {
                form.StartPosition = FormStartPosition.Manual;
                form.Location = this.PointToScreen(new Point(0, Height));
                form.ShowDialog();
            }
        }
    }
}

File: VB/ButtonEditWithPadding/PaddingButtonEdit.vb
Imports Microsoft.VisualBasic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Drawing

Namespace ButtonEditWithPadding
	Public Class PaddingButtonEdit
		Inherits ButtonEdit
		Shared Sub New()
			RepositoryItemPaddingButtonEdit.Register()
		End Sub
		Public Sub New()
		End Sub

		<DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
		Public Shadows ReadOnly Property Properties() As RepositoryItemPaddingButtonEdit
			Get
				Return TryCast(MyBase.Properties, RepositoryItemPaddingButtonEdit)
			End Get
		End Property

		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return RepositoryItemPaddingButtonEdit.EditorName
			End Get
		End Property


		Protected Overrides Sub OnClickButton(ByVal buttonInfo As EditorButtonObjectInfoArgs)
			ShowPopupForm()
			MyBase.OnClickButton(buttonInfo)
		End Sub
		Protected Overridable Sub ShowPopupForm()
			Using form As New Form()
				form.StartPosition = FormStartPosition.Manual
				form.Location = Me.PointToScreen(New Point(0, Height))
				form.ShowDialog()
			End Using
		End Sub
	End Class
End Namespace
